pkg/network/tracer/connection/kprobe/tracer.go

```diff
@@ -383,6 +404,27 @@ func (t *kprobeTracer) DumpMaps(maps ...string) (string, error) {
 	return t.m.DumpMaps(maps...)
 }
 
+// GetAllMapsNames returns the names of the maps managed by the manager
+func (t *kprobeTracer) GetAllMapsNames() []string {
+	var names []string
+
+	for _, m := range t.m.Maps {
+		names = append(names, m.Name)
+	}
+
+	return names
+}
+
+// GetAllProbesNames returns the names of the probes managed by the manager
+func (t *kprobeTracer) GetAllProbesNames() []string {
+	var names []string
+	for _, p := range t.m.Probes {
+		names = append(names, p.EBPFFuncName)
+	}
+
+	return names
+}
```

Comment:
Rather than requiring all "modules" to implement `GetAllMapsNames()` and `GetAllProbesNames()`, why don't you simply expose in your `telemetry` package a method that takes in an `ebpf.Manager` as argument? that way all the modules would have to simply call `bpfTelemetry.Register(self)` which is way less intrusive change IMO

Reply:
I think that would also obviate the need to propage these `ebpf.Map` types since you can do the map editing "yourself" inside the `Register` method

Reply:
Makes sense. I will add this change.